<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>serialForm</class>
 <widget class="QWidget" name="serialForm">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>810</width>
    <height>501</height>
   </rect>
  </property>
  <property name="minimumSize">
   <size>
    <width>810</width>
    <height>501</height>
   </size>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="windowTitle">
   <string>Form</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout" stretch="1,0">
   <property name="leftMargin">
    <number>5</number>
   </property>
   <property name="topMargin">
    <number>0</number>
   </property>
   <property name="rightMargin">
    <number>5</number>
   </property>
   <property name="bottomMargin">
    <number>5</number>
   </property>
   <item>
    <widget class="QGroupBox" name="serialReceiveAreaGroupBox">
     <property name="minimumSize">
      <size>
       <width>800</width>
       <height>238</height>
      </size>
     </property>
     <property name="title">
      <string>Receive Area</string>
     </property>
     <property name="checkable">
      <bool>false</bool>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="0" column="0">
       <layout class="QFormLayout" name="serialReceiveFormLayout">
        <item row="0" column="0" colspan="2">
         <widget class="SerialPortComboBox" name="serialComboBox">
          <property name="minimumSize">
           <size>
            <width>0</width>
            <height>0</height>
           </size>
          </property>
          <property name="maximumSize">
           <size>
            <width>170</width>
            <height>21</height>
           </size>
          </property>
         </widget>
        </item>
        <item row="1" column="0">
         <widget class="QLabel" name="serialBaudrateLabel">
          <property name="text">
           <string>Baudrate</string>
          </property>
         </widget>
        </item>
        <item row="1" column="1">
         <widget class="QComboBox" name="serialBaudrateComboBox">
          <item>
           <property name="text">
            <string>custom</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>110</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>300</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>4800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>9600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>19200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>38400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>57600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>76800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>115200</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>128000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>230400</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>256000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>460800</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>512000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>576000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>921600</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1000000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1500000</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2000000</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QLabel" name="serialStopBitLabel">
          <property name="text">
           <string>Stopbit</string>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QComboBox" name="serialStopBitComboBox">
          <item>
           <property name="text">
            <string>1</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>1.5</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>2</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QLabel" name="serialDataBitLabel">
          <property name="text">
           <string>Databit</string>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QComboBox" name="serialDataBitcomboBox">
          <item>
           <property name="text">
            <string>5</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>6</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>7</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>8</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="4" column="0">
         <widget class="QLabel" name="serialChecksumBitLabel">
          <property name="text">
           <string>Checksum</string>
          </property>
         </widget>
        </item>
        <item row="4" column="1">
         <widget class="QComboBox" name="serialChecksumBitComboBox">
          <item>
           <property name="text">
            <string>None</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Even</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Odd</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Mark</string>
           </property>
          </item>
          <item>
           <property name="text">
            <string>Space</string>
           </property>
          </item>
         </widget>
        </item>
        <item row="5" column="0">
         <widget class="QCheckBox" name="serialReceiveHexCheckBox">
          <property name="text">
           <string>Hex</string>
          </property>
         </widget>
        </item>
        <item row="5" column="1">
         <widget class="QCheckBox" name="serialReceiveTimestampCheckBox">
          <property name="text">
           <string>Timestamp</string>
          </property>
         </widget>
        </item>
        <item row="6" column="0" colspan="2">
         <widget class="QPushButton" name="serialConnectComPushButton">
          <property name="text">
           <string>Connect</string>
          </property>
         </widget>
        </item>
        <item row="7" column="0" colspan="2">
         <widget class="QPushButton" name="serialReceiveClearPushButton">
          <property name="text">
           <string>Clear</string>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="0" column="1">
       <widget class="QPlainTextEdit" name="serialReceivePlainTextEdit">
        <property name="undoRedoEnabled">
         <bool>false</bool>
        </property>
        <property name="readOnly">
         <bool>true</bool>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QTabWidget" name="serialTabWidget">
     <property name="currentIndex">
      <number>0</number>
     </property>
     <widget class="QWidget" name="serialTab_1">
      <attribute name="title">
       <string>Send Area</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="0" column="0" rowspan="3" colspan="4">
        <widget class="QPlainTextEdit" name="serialSendPlainTextEdit"/>
       </item>
       <item row="0" column="4">
        <widget class="QCheckBox" name="serialSendHexCheckBox">
         <property name="text">
          <string>Hex</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QPushButton" name="serialSendPushButton">
         <property name="text">
          <string>Send</string>
         </property>
        </widget>
       </item>
       <item row="2" column="4">
        <widget class="QPushButton" name="serialSendClearPushButton">
         <property name="text">
          <string>Clear</string>
         </property>
        </widget>
       </item>
       <item row="3" column="0">
        <widget class="QGroupBox" name="serialSoftFlowControlGroupBox">
         <property name="title">
          <string>Software Flow Control</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout">
          <item>
           <widget class="QCheckBox" name="serialSoftFlowControlCheckBox">
            <property name="text">
             <string>xon/xoff</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="2">
        <widget class="QGroupBox" name="serialSendRepeatGroupBox">
         <property name="title">
          <string>Repeat</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_4">
          <item row="0" column="0">
           <widget class="QCheckBox" name="serialSendRepeatCheckBox">
            <property name="text">
             <string/>
            </property>
           </widget>
          </item>
          <item row="0" column="2">
           <widget class="QLabel" name="serialSendRepeatUnitLabel">
            <property name="text">
             <string>ms</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="serialSendRepeatDurationLineEdit"/>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="3">
        <widget class="QGroupBox" name="serialSendLineFeedGroupBox">
         <property name="title">
          <string>New Line</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_3">
          <item>
           <widget class="QComboBox" name="serialSendLineFeedComboBox">
            <item>
             <property name="text">
              <string>None</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>CRLF</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>CR</string>
             </property>
            </item>
            <item>
             <property name="text">
              <string>LF</string>
             </property>
            </item>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="3" column="1">
        <widget class="QGroupBox" name="serialHardFlowControlGroupBox">
         <property name="title">
          <string>Hardware Flow Control</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_2">
          <item>
           <widget class="QCheckBox" name="serialHardFlowControlDSRDTRCheckBox">
            <property name="text">
             <string>dsr/dtr</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QCheckBox" name="serialHardFlowControlRTSCTSCheckBox">
            <property name="text">
             <string>rts/cts</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="serialTab_2">
      <attribute name="title">
       <string>File</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_5">
       <item row="0" column="1">
        <widget class="QGroupBox" name="serialFileSendGroupBox">
         <property name="title">
          <string>File Send</string>
         </property>
         <layout class="QHBoxLayout" name="horizontalLayout_4">
          <item>
           <widget class="QLabel" name="serialFileSendLabel">
            <property name="text">
             <string>File</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLineEdit" name="serialFileSendLineEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="serialFileSelectPushButton">
            <property name="text">
             <string>Select</string>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QPushButton" name="serialFileSendPushButton">
            <property name="text">
             <string>Send</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QGroupBox" name="serialFileSaveGroupBox">
         <property name="title">
          <string>File Save</string>
         </property>
         <layout class="QGridLayout" name="gridLayout_6">
          <item row="0" column="2">
           <widget class="QPushButton" name="serialFilePathSaveSelectPushButton">
            <property name="text">
             <string>Select</string>
            </property>
           </widget>
          </item>
          <item row="0" column="1">
           <widget class="QLineEdit" name="serialFilePathSaveLineEdit">
            <property name="readOnly">
             <bool>true</bool>
            </property>
           </widget>
          </item>
          <item row="0" column="0">
           <widget class="QLabel" name="serialFilePathSaveLabel">
            <property name="text">
             <string>Path</string>
            </property>
           </widget>
          </item>
          <item row="0" column="3">
           <widget class="QPushButton" name="serialFileSavePushButton">
            <property name="text">
             <string>Start</string>
            </property>
           </widget>
          </item>
         </layout>
        </widget>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
  </layout>
 </widget>
 <customwidgets>
  <customwidget>
   <class>SerialPortComboBox</class>
   <extends>QComboBox</extends>
   <header>widgets.serialPortComboBox.h</header>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections/>
</ui>
